honeybee_energy/simulation/output.py

```diff
@@ -0,0 +1,518 @@
+                try:
+                    frequency = ep_out_str[2] if ep_out_str[2] != '' else 'Hourly'
+                except IndexError:
+                    pass  # shorter output variable with default hourly frequencey
+        print(frequency)
+        # extract the summary_reports
+        reports = None
+        if summary_reports is not None:
+            reports = parse_idf_string(summary_reports, 'Output:Table:SummaryReports,')
+
+        return cls(outputs, frequency, include_sqlite, include_html, reports)
+
+    @classmethod
+    def from_dict(cls, data):
+        """Create a SimulationOutput object from a dictionary.
+
+        Args:
+            data: A SimulationOutput dictionary in following the format below.
+
+        .. code-block:: python
+
+            {
+            "type": "SimulationOutput",
+            "outputs": ['Zone Ideal Loads Supply Air Total Cooling Energy'],
+            "reporting_frequency": 'Annual',
+            "include_sqlite": False,
+            "include_html": True,
+            "summary_reports": ['AllSummary', 'AnnualBuildingUtilityPerformanceSummary']
+            }
+        """
+        assert data['type'] == 'SimulationOutput', \
+            'Expected SimulationOutput dictionary. Got {}.'.format(data['type'])
+        outputs = data['outputs'] if 'outputs' in data else None
+        frequency = data['reporting_frequency'] if \
+            'reporting_frequency' in data else 'Hourly'
+        sqlite = data['include_sqlite'] if 'include_sqlite' in data else True
+        html = data['include_html'] if 'include_html' in data else False
+        reports = data['summary_reports'] if 'summary_reports' in data else None
+        return cls(outputs, frequency, sqlite, html, reports)
+
+    def to_idf(self):
+        """Get EnergyPlus string representations of the SimulationOutput.
+
+        Returns:
+            table_style: An IDF OutputControl:Table:Style string for the simulation.
+            output_variables: A list of IDF Output:Variable strings for the requested
+                outputs. Will be None if no outputs have been requested.
+            summary_reports: An IDF Output:Table:SummaryReports string
+                listing the summary reports that are requested. Will be None
+                if no summary reports have not been requested.
+            sqlite: An IDF Output:SQLite string to request the SQLite file from
+                the simulation. Will be None if include_sqlite is False.
+            surfaces_list: An IDF Output:Surfaces:List string to ensure surface
+                information is written into the ultimate .eio file.
+        """
+        style = 'CommaAndHTML' if self.include_html else 'Comma'
+        table_style = generate_idf_string(
+            'OutputControl:Table:Style',
+            (style, 'JtoKWH'), ('column separator', 'unit conversion'))
+        output_variables = [self._output_to_idf(out_p) for out_p in self._outputs] if \
```

Comment:
I would use `set(self._outputs)` instead to remove possible duplicate outputs.

```suggestion
        output_variables = [self._output_to_idf(out_p) for out_p in set(self._outputs)] if \
```

Reply:
Let me think about this and possibly implement something tomorrow. If we are going to use sets, I'm not sure this is really the right place to do it.

Reply:
After thinking about it a bit, I decided it was best to use a `set` as the hidden attribute for `_outputs` instead of a `list`. This way, we avoid duplicate output names throughout the process of using the object rather than just waiting till it's written to an IDF or a dictionary to get rid of possible duplicates. The public attribute `outputs` is still a tuple so nothing has really changed on the interface end.